tests/BookReader.test.js

```diff
@@ -18,3 +37,88 @@ test('has registered view type selected events', () => {
 test('has registered fullscreen toggle event', () => {
   expect(BookReader.eventNames.fullscreenToggled).toBeTruthy();
 });
+
+test('checks cookie when initParams called', () => {
+  br.getResumeValue = jest.fn(() => 15);
+  br.urlReadFragment = jest.fn(() => '');
+
+  const params = br.initParams();
+  expect(br.getResumeValue).toHaveBeenCalledTimes(1);
+  expect(params.init).toBe(true);
+  expect(params.index).toBe(15);
+  expect(params.fragmentChange).toBe(true);
+});
+
+test('does not check cookie when initParams called', () => {
+  br.getResumeValue = jest.fn(() => null);
+  br.urlReadFragment = jest.fn(() => '');
+  br.options.enablePageResume = false;
+
+  const params = br.initParams();
+  expect(br.getResumeValue).toHaveBeenCalledTimes(0);
+  expect(params.init).toBe(true);
+  expect(params.index).toBe(0);
+  expect(params.fragmentChange).toBe(false);
+});
+
+test('overrides cookie if page fragment when InitParams called', () => {
+  br.getResumeValue = jest.fn(() => 15);
+  br.urlReadFragment = jest.fn(() => 'page/n4');
+  br.options.enablePageResume = true;
+
+  const params = br.initParams();
+  expect(br.getResumeValue).toHaveBeenCalledTimes(1);
+  expect(params.init).toBe(true);
+  expect(params.index).toBe(4);
+  expect(params.fragmentChange).toBe(true);
+});
+
+test('sets prevReadMode when init called', () => {
+  br.init();
+  expect(br.prevReadMode).toBeTruthy();
+});
+
+test('sets prevPageMode if initial mode is thumb', () => {
+  br.urlReadFragment = jest.fn(() => 'page/n4/mode/thumb');
+
+  br.init();
+  expect(br.prevReadMode).toBe(BookReader.constMode1up);
```

Comment:
question: was this the initial behavior showing a 1up mode if last captured mode was thumb or a new thing?

Reply:
It's already implemented in BookReader.prototype.drawLeafsThumbnail()
I noted with @see in the new function.
I'll be lazy and annotate the test in the next branch.
```
/**
 * Return mode or 1up if initial thumb
 * @param {number}
 * @see BookReader.prototype.drawLeafsThumbnail
 */
BookReader.prototype.getPrevReadMode = function(mode) {
```